src/app/modules/ix-feedback/feedback-dialog/feedback-dialog.component.ts

```diff
@@ -372,9 +369,14 @@ export class FeedbackDialogComponent implements OnInit {
 
     this.form.controls.token.valueChanges.pipe(
       filter((token) => !!token),
+      distinctUntilChanged(),
+      delay(10),
       untilDestroyed(this),
     ).subscribe((token) => {
       this.feedbackService.setOauthToken(token);
+      if (this.form.valid && !this.isLoading) {
+        this.onSubmit();
```

Comment:
So, user presses login to Jira button -> it updates token in a form -> this triggers submit in `addFormListeners`.
This won't be obvious to any future developer.
We should do it the right way by extracting common code from `OauthButtonComponent` into a service and calling it when normal submit button in this form is used.

Reply:
> We should do it the right way by extracting common code from OauthButtonComponent into a service and calling it when normal submit button in this form is used.

I am not sure if I get you. Can you explain the idea a bit more?

Reply:
Original Jira auth button doesn't work well in this form because it was not designed to be used in this scenario. Instead of a form control like in other forms, we need a generic button that shouldn't be responsible for Jira auth.
We need to put a normal button and start oAuth flow when it is pressed. To avoid copy-pasting existing code, we should reuse some of the code related to this from the OauthButtonComponent.

Reply:
If we use `ix-oauth-button` directly it allows us to remove `JiraOauthComponent` and keep it clear. Is it something we are looking for?

Reply:
Yeah, we can probably do that, it has that (loggedIn) output which should be more obvious to everyone.